mitmproxy/controller.py

```diff
@@ -239,52 +241,70 @@ class Reply(object):
     def __init__(self, obj):
         self.obj = obj
         self.q = queue.Queue()
-        # Has this message been acked?
-        self.acked = False
-        # Has the user taken responsibility for ack-ing?
-        self.taken = False
-        # Has a handler taken responsibility for ack-ing?
-        self.handled = False
+
+        self.state = "unhandled"  # "unhandled" -> "handled" -> "taken" -> "committed"
```

Comment:
I actually considered using Enum initially, but that requires an import of the enum class whereever we use this and things just get needlessly verbose. I changed `.state` to be readonly now, with modification only happening by the state transition functions below.